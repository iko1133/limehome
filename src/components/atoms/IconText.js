import React, { useContext } from "react";
import styled, { ThemeContext } from "styled-components/native";
import { MaterialIcons } from "@expo/vector-icons";

const Container = styled.View`
  align-items: center;
  flex-direction: row;
`;

const Text = styled.Text``;

/**
 * A text accompanied by icon
 * @param {string} text - the text displayed by the component
 * @param {string} iconName - the name of one of the MaterialIcons icon to be shown by the component
 * @param {object} textStyle - styling of the text
 * @param {object} style - styling of the component container
 * @param {float} iconSize - size of the icon (Defaults to 24)
 * @param {color} iconColor - color of the icon (Defaults to theme.SPECIAL_TEXT_COLOR)
 * @param {boolean} iconLeft - whether icon should be on the left or the right of the text
 */

const IconText = ({
  text,
  iconName,
  textStyle,
  style,
  iconSize = 24,
  iconColor,
  iconLeft = true,
}) => {
  const theme = useContext(ThemeContext);

  return (
    <Container style={style}>
      {iconLeft && (
        <MaterialIcons
          name={iconName}
          size={iconSize}
          color={iconColor ?? theme.SPECIAL_TEXT_COLOR}
        />
      )}
      <Text style={textStyle}>{text}</Text>
      {!iconLeft && (
        <MaterialIcons
          name={iconName}
          size={iconSize}
          color={iconColor ?? theme.SPECIAL_TEXT_COLOR}
        />
      )}
    </Container>
  );
};

export default IconText;
