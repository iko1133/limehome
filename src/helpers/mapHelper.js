// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
