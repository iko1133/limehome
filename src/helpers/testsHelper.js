import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
