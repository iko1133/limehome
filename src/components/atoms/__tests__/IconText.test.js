import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import IconText from "../IconText";

// Fixing vector icons problem with jest
jest.mock(
  "@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js",
  () => {
    return () => "";
  }
);

describe("Testing IconText", () => {
  it("Renders IconText with text 'Just Text' on it", () =>
    expect(
      renderer.create(WrapInThemeProvider(<IconText text="Just Text" />))
    ).toMatchSnapshot());

  it("Renders IconText with text 'Text with location pin' on it with location pin icon on left", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <IconText text="Text with location pin" iconName="location-pin" />
        )
      )
    ).toMatchSnapshot());

  it("Renders IconText with text 'Text with star icon on right' on it with star icon on the right", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <IconText
            text="Text with star icon on right"
            iconName="star"
            iconLeft={false}
          />
        )
      )
    ).toMatchSnapshot());
});
