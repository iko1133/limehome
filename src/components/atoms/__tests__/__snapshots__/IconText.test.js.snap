// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing IconText Renders IconText with text 'Just Text' on it 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {},
      ]
    }
  >
    Just Text
  </Text>
</View>
`;

exports[`Testing IconText Renders IconText with text 'Text with location pin' on it with location pin icon on left 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {},
      ]
    }
  >
    Text with location pin
  </Text>
</View>
`;

exports[`Testing IconText Renders IconText with text 'Text with star icon on right' on it with star icon on the right 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {},
      ]
    }
  >
    Text with star icon on right
  </Text>
  <Text />
</View>
`;
